src/probe/probe.c

```diff
@@ -643,12 +626,20 @@ static int probe_gen_header(uint32_t buffer_id, uint32_t size,
 	header->format = format;
 	header->timestamp_low = (uint32_t)timestamp;
 	header->timestamp_high = (uint32_t)(timestamp >> 32);
-	header->checksum = 0;
 	header->data_size_bytes = size;
-
+#if CONFIG_IPC_MAJOR_4
+	/* calc checksum to check validation by probe parse app */
+	*checksum = header->sync_word +
+		    header->buffer_id  +
+		    header->format +
+		    header->timestamp_high +
+		    header->timestamp_low +
+		    header->data_size_bytes;
+#else
 	/* calc crc to check validation by probe parse app */
-	crc = crc32(0, header, sizeof(*header));
-	header->checksum = crc;
+	header->crc = 0;
```

Comment:
I don't think you renamed `.checksum` to `.crc` in this patch

Reply:
Renamed in ipc3 header back to crc